coq/coq-system.el

```diff
@@ -140,7 +140,10 @@ Interactively (with INTERACTIVE-P), show that number."
 (defun coq-autodetect-help (&optional interactive-p)
   "Record the output of coqotp -help in `coq-autodetected-help'."
   (interactive '(t))
-  (setq coq-autodetected-help (coq-callcoq "-help")))
+  (let ((coq-output (coq-callcoq "-help")))
+    (if coq-output
+        (setq coq-autodetected-help coq-output)
+      (setq coq-autodetected-help ""))))
```

Comment:
I think this is the same as ` (setq coq-autodetected-help (or (coq-callcoq "-help") ""))`?
Are we sure there were no other callers that might be affected by the callcoq change?

Reply:
I don't understand the question. `coq-callcoq` is not modified, only this call to it. The only other call (10 lines above) is already protected.

Reply:
I hadn't checked all the callers.  The bug that this PR fixes was introduced by modifying callcoq, so it seemed natural to make the fix there too.

Reply:
I see. I think this is OK like this.

Reply:
> I hadn't checked all the callers. The bug that this PR fixes was introduced by modifying callcoq, so it seemed natural to make the fix there too.

A simple grep shows there are only 2 uses of `coq-callcoq`. The first one protected itself already against a return value nil for a long time. Further, returning the empty string from `coq-callcoq` when `coq-prog-name` cannot be found --which was the case before PR #526-- is a design error IMO, especially, because there has always been a code path that returned nil. I considered all this before preparing this PR and therefore decided to protect the only remaining use of `coq-callcoq`.